pkg/kubernetes/client.js

```diff
@@ -680,6 +697,34 @@ define([
         });
 
         /*
+         * add_watcher:
+         * @type: The type of watch to add
+         *
+         * Returns a Promise that if successfull will
+         * be a KubernetesWatch object for the given type.
+         * The same object will be available at self.watches[type]
+         */
+        self.add_watcher = function add_watcher (type) {
+            var dfd = $.Deferred();
+
+            self.connect()
```

Comment:
I don't think this function needs to wait to connect or anything like that.

I would suggest just adding to self.watches immediately. Why do we need to wait for connect to happen? But we do need to see if already connected ... and if so call watch.start() here.